integration/tests/bft/gossipsub-signature/signature-requirement/orchestrator.go

```diff
@@ -0,0 +1,171 @@
+		codec:                      unittest.NetworkCodec(),
+		unauthorizedEventsReceived: atomic.NewInt64(0),
+		authorizedEventsReceived:   atomic.NewInt64(0),
+		unauthorizedEvents:         make(map[flow.Identifier]*insecure.EgressEvent),
+		authorizedEvents:           make(map[flow.Identifier]*insecure.EgressEvent),
+		authorizedEventReceivedWg:  sync.WaitGroup{},
+		attackerVNNoMsgSigning:     attackerVNNoMsgSigning,
+		attackerVNWithMsgSigning:   attackerVNWithMsgSigning,
+		victimENID:                 victimEN,
+	}
+
+	return orchestrator
+}
+
+// HandleEgressEvent implements logic of processing the outgoing (egress) events received from a corrupted node. This attack orchestrator
+// simply passes through messages without changes to the orchestrator network.
+func (s *SignatureValidationAttackOrchestrator) HandleEgressEvent(event *insecure.EgressEvent) error {
+	lg := s.logger.With().
+		Hex("corrupt_origin_id", logging.ID(event.CorruptOriginId)).
+		Str("channel", event.Channel.String()).
+		Str("protocol", event.Protocol.String()).
+		Uint32("target_num", event.TargetNum).
+		Strs("target_ids", logging.IDs(event.TargetIds)).
+		Str("flow_protocol_event", logging.Type(event.FlowProtocolEvent)).Logger()
+
+	err := s.orchestratorNetwork.SendEgress(event)
+	if err != nil {
+		lg.Error().Err(err).Msg("could not pass through egress event")
+		return err
+	}
+
+	lg.Info().Str("event_id", event.FlowProtocolEventID.String()).Msg("egress event passed through successfully")
+	return nil
+}
+
+// HandleIngressEvent implements logic of processing the incoming (ingress) events to a corrupt node.
+// This handler will track any unauthorized messages that are expected to be blocked by libp2p for message
+// signature validation failure. It also tracks all the authorized messages that are expected to be delivered to the node.
+func (s *SignatureValidationAttackOrchestrator) HandleIngressEvent(event *insecure.IngressEvent) error {
+	lg := s.logger.With().
+		Hex("origin_id", logging.ID(event.OriginID)).
+		Str("channel", event.Channel.String()).
+		Str("corrupt_target_id", fmt.Sprintf("%v", event.CorruptTargetID)).
+		Str("flow_protocol_event", fmt.Sprintf("%T", event.FlowProtocolEvent)).Logger()
+
+	// Track any unauthorized events that are received by the corrupted node that has disabled message signing.
+	// These messages should have been rejected by libp2p.
+	if _, ok := s.unauthorizedEvents[event.FlowProtocolEventID]; ok {
+		s.unauthorizedEventsReceived.Inc()
+		lg.Warn().Str("event_id", event.FlowProtocolEventID.String()).Msg("unauthorized ingress event received")
+	}
+
+	// track all authorized events sent during test
+	if expectedEvent, ok := s.authorizedEvents[event.FlowProtocolEventID]; ok {
+		// ensure event received intact no changes have been made to the underlying message
+		require.Equal(s.t, expectedEvent.FlowProtocolEvent.(*messages.ChunkDataRequest), event.FlowProtocolEvent.(*messages.ChunkDataRequest))
```

Comment:
Please also check the other fields as well; e.g., `Channel`.

Reply:
https://github.com/onflow/flow-go/pull/3689/commits/e7e31394c2ade53550cad8a63f2da09eb275638b